src/uri.rs

```diff
@@ -61,12 +67,12 @@ impl<T: DeserializeOwned> Uri<T> {
         match response {
             Ok(response) => match response.status() {
                 200..=299 => Ok(serde_json::from_reader(response.into_reader())?),
-                status => Err(Error::HttpError(status, response.status_text().to_string())),
+                status => Err(Error::HttpError(status, response.status_text().to_string()).into()),
```

Comment:
Hmm.. this allocation here seems unnecessary, aren't these messages standardized according to the status code?

Couldn't we use https://docs.rs/httpstatus/0.1.2/httpstatus/enum.StatusCode.html instead and and save on this allocation and on the string in our error type?

Reply:
Done.